### PR TITLE
feature-8784:Add type year drop down custom from field

### PR DESCRIPTION
**Pull request by khangon**
Fixes https://github.com/fossasia/open-event-frontend/issues/8784

#### Short description of what this resolves:
- Add type year drop down custom from field

#### Changes proposed in this pull request:
- Add type year drop down custom from field

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by norbusan** [comment]
Same problem with your other PR, CI failing.

**Comment by khangon** [comment]
Hi @cweitat, @norbusan 
As expected, we have added Year to the drop-down.
Please help me check and verify.
Khang On
year

**Comment by cweitat** [comment]
@khangon fix codacy pls thanks

File: app/utils/dictionary/year-list.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+let year: number = new Date().getFullYear();
+const earliestYear = 1970;
```
Comment (norbusan) [inline app/utils/dictionary/year-list.ts:2]: The issue https://github.com/fossasia/open-event-frontend/issues/8784 lists 1930 as the start year?

**Review by norbusan — COMMENTED** [review]
Although the issue https://github.com/fossasia/open-event-frontend/issues/8784 specifies 1930-current year, I am not happy with having a fixed list of years.

**Comment by lthanhhieu** [comment]
> 

Hi @norbusan,
When we started to implement this ticket, the description did not mention the start year and end year, so we decided to use 1970 as the start due to the javascript provided.
For the current year, which is the end year, I think we also discussed creating a new ticket to change the end year from the current year to the feature year. In that case, could you please suggest how many years we will add to the feature? (ex: 1930-?)

Hi @mariobehling,
Please notify us if you change the description in the ticket.

**Comment by norbusan** [comment]
The change to 1930 should be quick, 1 character change ;-)

Concerning future year, yes, that should go into a new ticket.

File: app/utils/dictionary/year-list.ts (hunk 1 above)
Comment (khangon) [reply app/utils/dictionary/year-list.ts]: Hi @norbusan,
I changed the starting year to 1930.
Please help me check.

**Comment by mariobehling** [comment]
Please resolve conflicts with your other PR. Thank you!

**Review by cweitat — CHANGES_REQUESTED** [review]
Conflicts

**Pull request merged**